### PR TITLE
Make all Grid column groups required

### LINKED ISSUE #2282: Should we make all Grid column groups required?
Based on this comment https://github.com/bbc/psammead/pull/2213#discussion_r329520106 should we consider making all Grid column groups required?

If we don't define a particular group's columns then it defaults to full width. This may not be desirable behaviour and could look broken if we have forgotten to provide a group with columns.

The benefit of the current functionality is if we want items to span the full width of the page we can either not pass in those groups or you can configure the groups to span the full width. The latter would render a lot of CSS to do what the former does by default. So as long as we are careful about the group column set up then we could potentially save on CSS.

### PR DESCRIPTION
**Pull request by EinsteinNjoroge**
Resolves #2282 

**Overall change:** _Make all Grid column groups required._

**Code changes:**

- _Make all Grid column groups required._
- _Update documentation._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-grid/README.md [hunk 1]
```diff
@@ -22,19 +22,42 @@ Psammead Grid is a component that you can use to set out column-based layouts us
...
+    columns={{
+      group0: 5,
+      group1: 5,
+      group2: 5,
+      group3: 5,
+      group4: 5,
+      group5: 5,
+    }}
+  >
+    This spans 5 out of 6 columns for all width
```
Comment (sareh) [inline packages/components/psammead-grid/README.md]:
```suggestion
    This spans 5 out of 6 columns for all widths
```

File: packages/components/psammead-grid/README.md [hunk 2]
```diff
@@ -53,25 +76,72 @@ When we refer to `group3` in this component, we're referring to the breakpoint w
...
 
 ```jsx
 import Grid from '@bbc/psammead-grid';
 
 const MyComponent = () => (
-  <Grid columns={{ group4: 8 }}>
-    <Grid item columns={{ group4: 6 }}>
+  <Grid
+    columns={{
+      group0: 6,
```
Comment (sareh) [inline packages/components/psammead-grid/README.md]: To keep this as a simple example, it'd be good to ensure it matches the description above - so has groups 0-5 as `8` on the outer grid and `6` / `2` for the relevant Grid items.

File: packages/components/psammead-grid/README.md [hunk 3]
```diff
@@ -136,19 +213,77 @@ Note that here, any time you use `<Grid>` that generates a new grid. The total n
 import Grid from '@bbc/psammead-grid';
 
 const MyComponent = () => (
-  <Grid columns={{ group2: 6, group3: 6 }}>
-    <Grid columns={{ group2: 6, group3: 6 }}>
-      <Grid item columns={{ group2: 6, group3: 3 }}>
+  <Grid
+    columns={{
+      group0: 4,
```
Comment (sareh) [inline packages/components/psammead-grid/README.md]: To ensure this example is easier to follow, please keep these columns having the same values. e.g. `group0: 6, group1: 6, group2: 6, group3: 6, group4: 6, group5: 6`. That way, if someone was to use this example, the output would look just like the screenshot shown below: https://github.com/bbc/psammead/pull/2458/files#diff-afe1f4233782372163743bf32a6bc5eeR294

**Review by sareh — CHANGES_REQUESTED** [review]
Looks good so far. Thanks for updating the Readme. I've added some comments about the example values below.

File: packages/components/psammead-grid/README.md [hunk 4]
```diff
@@ -55,56 +78,53 @@ When we refer to `group3` in this component, we're referring to the breakpoint w
...
-      <p>Item 4 - Paragraph that spans 2 out of 8 columns through group4</p>
-    </Grid>
   </Grid>
 );
 ```
 
 Image of this example when viewed at within `group4` (from 1008px to 1279px). When you look at this example at any other breakpoint, each of the four items fill up the full width of the row.
 ![8 column grid with items 6 2 2 2 with labels](packages/components/psammead-grid/documentation/8-column-grid-with-items-6-2-2-2-with-labels.png)
 
-#### <a name="multiple-groups">Setting values for multiple groups</a>
```
Comment (DenisHdz) [inline packages/components/psammead-grid/README.md:84]: Why has this example been removed?

File: packages/components/psammead-grid/README.md [hunk 5]
```diff
@@ -55,56 +78,53 @@ When we refer to `group3` in this component, we're referring to the breakpoint w
...
-      <p>
-        Paragraph - for group 3 spans 6/6 columns, for group 4 spans 2/8 columns
-      </p>
-    </Grid>
-  </Grid>
-);
-```
-
-Screenshot of this example without GEL Gutters and GEL Margins
-![Screenshot of this example without GEL Gutters and GEL Margins](packages/components/psammead-grid/documentation/screenshot-example-without-gelMargins-and-gelGutters.png)
```
Comment (sareh) [inline packages/components/psammead-grid/README.md:106]: It'd be good to keep this example without GEL gutters and margins here. With a fixed relative link `documentation/screenshot-example-without-gelMargins-and-gelGutters.png`

Comment (sareh) [reply packages/components/psammead-grid/README.md:106]: Thanks for the other updates! They look great.

File: packages/components/psammead-grid/README.md (hunk 4 above)
Comment (EinsteinNjoroge) [reply packages/components/psammead-grid/README.md:84]: All grid examples in this documentation now have multiple column groups, this example is no longer needed.

File: packages/components/psammead-grid/README.md (hunk 5 above)
Comment (EinsteinNjoroge) [reply packages/components/psammead-grid/README.md:106]: This screenshot is included [here](https://github.com/bbc/psammead/pull/2458/files#diff-afe1f4233782372163743bf32a6bc5eeR208)

**Review by sareh — APPROVED** [review]
Thanks for the Readme updates. It looks good.

**Review by thekp — APPROVED** [review]

**Pull request merged**